requirements-dev.txt

```diff
@@ -0,0 +1,12 @@
+# Requirements for development environment. These are not needed for CI.
```

Comment:
I know we use `poetry` all around the services to manage python dependencies. I think we can use it here as well. 
I wonder if we also can use poetry's virtual env instead of devcontainer to manage this environment? I don't have experience with VSCode devcontainer, though

Reply:
True this can be done in a poetry virtualenv as well, but I feel this is more reproducible and more similar to the CI environment. Longer story: I was getting a bunch of segfaults in python libs when running the SDK on OSX, and had to reinstall python on two different occasions, so decided to develop in a docker container instead.